recipes/openassetio/all/test_package/conanfile.py

```diff
@@ -0,0 +1,76 @@
+import pathlib
+
+from conan import ConanFile
+from conan.tools.microsoft import is_msvc
+from conan.tools.build import can_run
+from conan.tools.cmake import cmake_layout, CMake, CMakeToolchain, CMakeDeps
+from conan.tools.files import get
+
+
+class TestPackageConan(ConanFile):
+    settings = "os", "arch", "compiler", "build_type"
+    generators = "CMakeDeps", "CMakeToolchain", "VirtualRunEnv"
+    test_type = "explicit"
+
+    def build_requirements(self):
+        self.tool_requires("cmake/3.25.1")
+
+    def requirements(self):
+        self.requires("openssl/1.1.1s")
+        self.requires(self.tested_reference_str)
+
+        if "with_python" in self.options["openassetio"] and not self.options["openassetio"].with_python:
+            python_version = None
+        elif "python_version" in self.options["openassetio"]:
+            python_version = self.options['openassetio'].python_version
+        else:
+            python_version = "3.9.7"
+
+        if python_version is not None:
+            self.requires(f"cpython/{python_version}")
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    def generate(self):
+        get(self, **self.conan_data["sources"][self.dependencies["openassetio"].ref.version], destination=self.source_folder, strip_root=True)
```

Comment:
As mentioned this is a blocker.

There should be a cpp with less then 100 of code to test the library links and can be run on the same config

Reply:
Removed the external fetch of test project and bundled a simplified version instead in 3fba84f3.  All together comes to less than 100 lines of code.